misk/src/main/kotlin/misk/tasks/RepeatedTaskQueue.kt

```diff
@@ -76,16 +76,30 @@ class RepeatedTaskQueue @VisibleForTesting internal constructor(
       // task requests rescheduling
       taskExecutor.submit {
         val result = task()
-        if (result.status != Status.NO_RESCHEDULE) schedule(result.nextDelay, task)
+        // Reschedule using enqueue so as to not repeatedly wrap the task in try-catch blocks
+        if (result.status != Status.NO_RESCHEDULE) enqueue(result.nextDelay, task)
       }
     }
   }
 
   /**
    * Schedules a task to run repeatedly after an initial delay. The task itself determines the
-   * next execution time
+   * next execution time. Provide an optional retryDelayOnFailure parameter to determine when
+   * the job should be retried in the case of an unhandled exception by the client
    */
-  fun schedule(delay: Duration, task: () -> Result) {
+  fun schedule(delay: Duration, retryDelayOnFailure: Duration? = null, task: () -> Result) {
+    val wrappedTask: () -> Result = {
+      try {
+        task()
+      } catch (th: Throwable) {
```

Comment:
I think you want `Exception`? `Errors` aren't usually meant to be caught

Reply:
I think you want `Throwable`. if some library accidentally throws an `AssertionError` that will kill the scheduling thread.